### ISSUE TITLE
removeObject3D does not de-reference object3D.el

### ISSUE THREAD

**Issue by gfodor** [aframevr/aframe#4637]
**Description:**

I've been tracking down memory leaks and I noticed one stemming from the fact that if you call `el.setObject3D("mesh", mesh)` and then `el.removeObject3D("mesh")`, then `mesh.el` still references the A-Frame element. so if mesh is pinned to memory, so is the entire aframe subtree.

- A-Frame Version: 1.0.3
- Platform / Device: Linux/Chrome
- Reproducible Code Snippet or URL:

**Comment by arpu** [comment]
hi @gfodor  

how do you think this could be solved ? 
deleting this in https://github.com/aframevr/aframe/blob/master/src/core/a-entity.js#L163 ?

**Comment by gfodor** [comment]
no, I think what needs to happen is `obj.el = null;` - otherwise there is a dangling reference on the Object3D to the a-frame element. however, this change could be problematic if the obj has been added into the object3D map multiple times for different keys, if that is a valid configuration then `el = null` would have to happen if and only if it is no longer represented in the map.

**Comment by dmarcos** [comment]
When calling `el.removeObject3D("mesh")` we do `delete this.object3DMap['mesh'];` if there are no more references to the mesh in the code I think it should be garbaged collected and the references to `el` also removed? I might be wrong.
